libs/experimental/langchain_experimental/llms/ollama_functions.py

```diff
@@ -293,10 +293,9 @@ def _generate(
                     "matching function in `functions`."
                 )
             del kwargs["function_call"]
-        elif not functions:
-            functions.append(DEFAULT_RESPONSE_FUNCTION)
         if _is_pydantic_class(functions[0]):
             functions = [convert_to_ollama_tool(fn) for fn in functions]
+        functions.append(DEFAULT_RESPONSE_FUNCTION)
```

Comment:
why do we always append the default response function? could we propose this in a separate PR if it's needed?

Reply:
Default conversation prompt added to handle cases where available tools are not appropriate for handling the question and to allow the LLM to respond to inputs by ignoring the tools.
Like available tools contain weather tool but question asked is "what is capital of France".
I will see if I can identify another approach to handle this in a more idiomatic way.